kong/cluster_events.lua

```diff
@@ -0,0 +1,334 @@
+local utils = require "kong.tools.utils"
+
+
+local ngx_debug = ngx.config.debug
+local DEBUG     = ngx.DEBUG
+local ERR       = ngx.ERR
+local max       = math.max
+local type      = type
+local pcall     = pcall
+local insert    = table.insert
+local ngx_log   = ngx.log
+local ngx_now   = ngx.now
+local timer_at  = ngx.timer.at
+
+
+local NODE_ID_KEY    = "cluster_events:id"
+local POLL_LOCK_KEY  = "cluster_events:poll"
+local CURRENT_AT_KEY = "cluster_events:at"
+
+
+local POLL_LOCK_EXPTIME = 60 * 5       -- 5 mins
```

Comment:
should we derive this from something configurable, such as `db_update_frequency`, rather than a hardcoded value?